PDE/advection1reverse.m

```diff
@@ -0,0 +1,103 @@
+function pde = advection1reverse
+%1D Test case of the 1D advection equation with
+% the direction of motion reversed from advection1.m.
+% Code designed to test inhomogeneous Dirichlet BC in conjunction with
+% advection1.m
+% Equation to be solved is 
+% df/dt = df/dx + source(x)
+% Run with
+%
+% explicit
+% asgard(advection1reverse)
+% asgard(advection1reverse,'lev',4,'deg',3,'implicit',false)
+%
+% implicit
+% asgard(advection1reverse,'implicit',true)
+% asgard(advection1reverse,'implicit',true,'CFL',0.01)
+
+%% Setup the dimensions
+% 
+% Here we setup a 1D problem (x) on [a,b]
+
+dim_x.domainMin = 0; %a
+dim_x.domainMax = pi; %b
+dim_x.init_cond_fn = @(x,p,t) cos(x);
+
+%%
+% Add dimensions to the pde object
+% Note that the order of the dimensions must be consistent with this across
+% the remainder of this PDE.
+
+pde.dimensions = {dim_x};
+num_dims = numel(pde.dimensions);
+
+%Inhomogeneous Dirichlet condition on one side of the domain
+
+BC_Func_Left = @(x) x.*0 + 1;
+BC_Func_Right = @(x) x.*0 - 1;
+%BCFunc_t = @(t) soln_t(t);
+
+BCL_fList = { ... 
+     @(x,p,t) BC_Func_Left(x), ... %replace x with a
+     @(t,p)  t.*0 + 1 %boundary condition for time, usually 1
+     };
+
+BCR_fList = { ... 
+     @(x,p,t) BC_Func_Right(x), ... %replace x with b
+     @(t,p)  t.*0 + 1 %boundary condition for time, usually 1
+     };
+
+%Setup the terms of the PDE
+% Here the term is df/dx, which is opposite the direction in advection1.m
+
+g1rev = @(x,p,t,dat) x.*0 + 1;
```

Comment:
remove the `rev` part of the `g1rev` variable name

Reply:
g1rev has been changed to g1 in advection1reverse.m